Dockerfile

```diff
@@ -1,42 +1,82 @@
-RUN apk update && apk add --no-cache wget curl git yarn build-base gcc abuild binutils binutils-doc gcc-doc zip
+# 编译前端代码
+WORKDIR /source/assets
 
-WORKDIR /cloudreve_builder
-RUN git clone --recurse-submodules https://github.com/cloudreve/Cloudreve.git
+# 允许通过 `docker build --build-args YARN_REGISTRY=https://registry.npmmirror.com` 覆盖默认仓库地址
+#
+# 常用的仓库地址:
+#   npm -----  https://registry.npmjs.org
+#   cnpm ----  http://r.cnpmjs.org
+#   taobao --  https://registry.npmmirror.com
+#   nj ------  https://registry.nodejitsu.com
+#   rednpm --  http://registry.mirror.cqupt.edu.cn
+#   skimdb --  https://skimdb.npmjs.com/registry
+#   yarn ----  https://registry.yarnpkg.com
+ARG YARN_REGISTRY=https://registry.yarnpkg.com
 
-# build frontend
-WORKDIR /cloudreve_builder/Cloudreve/assets
-RUN yarn install --network-timeout 1000000
-RUN yarn run build && find . -name "*.map" -type f -delete
+# 允许通过 `docker build --build-args GOPROXY=https://goproxy.cn` 添加 go mod 代理
+ARG GOPROXY=""
 
-# build backend
-WORKDIR /cloudreve_builder/Cloudreve
-RUN zip -r - assets/build >assets.zip
-RUN tag_name=$(git describe --tags) \
-    && export COMMIT_SHA=$(git rev-parse --short HEAD) \
-    && go build -a -o cloudreve -ldflags " -X 'github.com/HFO4/cloudreve/pkg/conf.BackendVersion=$tag_name' -X 'github.com/HFO4/cloudreve/pkg/conf.LastCommit=$COMMIT_SHA'"
+# 暂不确定未来 alpine 内的 Node 版本是否影响最终编译结果, 故暂时增加打印输出
+RUN set -ex \
+    && echo "Node Version: $(node -v)" \
+    && sed -i -e "s@https://registry.yarnpkg.com@${YARN_REGISTRY}@g" yarn.lock \
+    && yarn install --network-timeout 1000000 \
+    && yarn run build \
+    && find ./build -name "*.map" -type f -delete
 
+# 编译后端代码
+WORKDIR /source
 
-# build final image
-FROM alpine:latest
+# assets.zip: 用于 go:embed 嵌入, 不可修改文件名
+RUN set -ex \
+    && zip -r - assets/build > assets.zip \
+    && go build -trimpath -o /cloudreve.bin -ldflags \
+            "-w -s \
+             -X github.com/cloudreve/Cloudreve/v3/pkg/conf.BackendVersion=$(git describe --tags) \
+             -X github.com/cloudreve/Cloudreve/v3/pkg/conf.LastCommit=$(git rev-parse --short HEAD)"
 
-WORKDIR /cloudreve
 
-RUN apk update && apk add --no-cache tzdata
+# 多阶段构建: 构建最终镜像 
+FROM alpine
 
-# we using the `Asia/Shanghai` timezone by default, you can do modification at your will
-RUN cp /usr/share/zoneinfo/Asia/Shanghai /etc/localtime \
-    && echo "Asia/Shanghai" > /etc/timezone
+# 允许通过 `docker build --build-args TIMEZONE=Africa/Abidjan` 覆盖默认时区
+ARG TZ="Asia/Shanghai"
 
-COPY --from=cloudreve_builder /cloudreve_builder/Cloudreve/cloudreve ./
+# netgo 兼容修复
+# set up nsswitch.conf for Go's "netgo" implementation
+# - https://github.com/golang/go/blob/go1.9.1/src/net/conf.go#L194-L275
+# - docker run --rm debian:stretch grep '^hosts:' /etc/nsswitch.conf
+RUN [ ! -e /etc/nsswitch.conf ] && echo 'hosts: files dns' > /etc/nsswitch.conf
 
-# prepare permissions and aria2 dir
-RUN chmod +x ./cloudreve && mkdir -p /data/aria2 && chmod -R 766 /data/aria2
```

Comment:
Have you tested if this works without grant permission for the remote download directory? If so, please provide the screenshot or something that could prove the remote download actually works